lib/streamlit/util.py

```diff
@@ -126,15 +126,17 @@ def index_(iterable, x) -> int:
     -------
     int
     """
-
+    if x is None:
+        return 0
+    
     for i, value in enumerate(iterable):
         # https://stackoverflow.com/questions/588004/is-floating-point-math-broken
         # https://github.com/streamlit/streamlit/issues/4663
-        if isinstance(value, np.float64) or isinstance(value, float):
+        if x == value:
+            return i
+        elif isinstance(value, np.float64) or isinstance(value, float):
```

Comment:
nit: slightly out of scope of your change, but while we're making changes to this function, this could be more cleanly rewritten as just `isinstance(value, float)` because `np.float64` subclasses `float`